File: python/desafio3.py
def solution(statues):
    i = 0;
    while(i < len(statues)-1):
        if(statues[i+1] < statues[i]):
            statues.append(statues[i]);
            statues.pop(i);
        else:
            i += 1;
    i= 0;
    result = 0;    
    while(i < len(statues)-1):
        if(statues[i+1] - statues[i] == 1):
            i += 1;
        else:
            result += statues[i+1] - statues[i] - 1;
            i += 1;
    return result;     

File: php/desafio3.php
function solution($statues) {
    $i = 0;
    while($i < count($statues) - 1){
        if($statues[$i+1] < $statues[$i]){
            array_push($statues, $statues[$i]);
            array_splice($statues, $i, 1);
        }else {
            $i++;
        }
    }
    $i = 0;
    $result = 0;
    while($i < count($statues) - 1){
        if($statues[$i+1] - $statues[$i] == 1){
            $i++;
        }else {
            $result = $result + ($statues[$i+1] - $statues[$i] - 1);
            $i++;
        }
    }
    return $result;
}
